### ISSUE TITLE
[BUG] Cannot register on fresh installation

### ISSUE THREAD

**Issue by reanim8ed** [standardnotes/standalone#72]
After fresh server setup I tried to register new user in the app.standardnotes.org platform with my custom server address in the advanced options form. Immedietly after POST request, I get expired login screen but loging in with same credentials doesnt work.
In server logs I can see failed attempts logged. But if I check the the users table in mysql container - it's still empty. 

I tried to use these new credentials on default sync server - and I'm logged in.
It seems that registration form doesnt care about custom server address account is not registered on the custom server but in api.standardnotes.com server instead.

**Comment by stevensiebe** [comment]
A similar error for me, using a fresh install on a fresh server
`A server error occurred while trying to register. Please try again.`
Trying again shows
`An existing registration request is already in progress.`

**Comment by Nicicalu** [comment]
Same here!

**Comment by dasTor** [comment]
i have updated my backend yesterday, client was freshly downloaded yesterday too.
i can see in the error_console of the client (mac os) that even though custom sync server is set,
the registration request is sent to standardnotes.com

**Comment by moughxyz** [comment]
Ah, indeed. Looks like a recent regression caused by a refactor of our registration handler. Will be fixed by https://github.com/standardnotes/snjs/pull/771.

**Comment by Nicicalu** [comment]
> Ah, indeed. Looks like a recent regression caused by a refactor of our registration handler. Will be fixed by [standardnotes/snjs#771](https://github.com/standardnotes/snjs/pull/771).

🙂 thanks!

**Comment by Nicicalu** [comment]
> Ah, indeed. Looks like a recent regression caused by a refactor of our registration handler. Will be fixed by [standardnotes/snjs#771](https://github.com/standardnotes/snjs/pull/771).

Does this mean that it should work, the next time I update the server? Or do I have to wait for a new image of a container or a new release?

**Comment by Nicicalu** [comment]
@moughxyz

**Comment by carlox97** [comment]
@moughxyz I'm having the same problem: I self-hosted the docker standalone server and upon the first login after registering I get asked a 2FA code which prevents me from logging in.

**Comment by Nicicalu** [comment]
> @moughxyz I'm having the same problem: I self-hosted the docker standalone server and upon the first login after registering I get asked a 2FA code which prevents me from logging in.

Did you run `./server.sh update` on your server?

**Comment by carlox97** [comment]
> > @moughxyz I'm having the same problem: I self-hosted the docker standalone server and upon the first login after registering I get asked a 2FA code which prevents me from logging in.
> 
> Did you run `./server.sh update` on your server?

Yes, I did. I'm trying to host a local istance (no https) before going public.

`git clone --single-branch --branch main https://github.com/standardnotes/standalone.git
cd standalone`
`./server.sh setup`
`sed -i "s/auth_jwt_secret/$(openssl rand -hex 32)/g" .env
sed -i "s/secret/$(openssl rand -hex 32)/g" docker/auth.env
sed -i "s/legacy_jwt_secret/$(openssl rand -hex 32)/g" docker/auth.env
sed -i "s/secret_key/$(openssl rand -hex 32)/g" docker/auth.env
sed -i "s/server_key/$(openssl rand -hex 32)/g" docker/auth.env`
`cd standalone`
`./server.sh update`

Status looks good, everything running, healtcheck OK.

I tried to register from iOS, Android, Windows using "http://10.0.0.100:3000", I got a generic error at times, upon clicking register. Cannot register with another account because "another registration in progress" (or something in those lines). Upon logging in from mentioned platforms I get prompted the 2FA code, which I don't have, obviously, and at the same time the logs show:

 
`auth-standalone                      | query: SELECT `user`.`uuid` AS `user_uuid`, `user`.`version` AS `user_version`, `user`.`email` AS `user_email`, `user`.`pw_nonce` AS `user_pw_nonce`, `user`.`encrypted_server_key` AS `user_encrypted_server_key`, `user`.`server_encryption_version` AS `user_server_encryption_version`, `user`.`kp_created` AS `user_kp_created`, `user`.`kp_origination` AS `user_kp_origination`, `user`.`pw_cost` AS `user_pw_cost`, `user`.`pw_key_size` AS `user_pw_key_size`, `user`.`pw_salt` AS `user_pw_salt`, `user`.`pw_alg` AS `user_pw_alg`, `user`.`pw_func` AS `user_pw_func`, `user`.`encrypted_password` AS `user_encrypted_password`, `user`.`created_at` AS `user_created_at`, `user`.`updated_at` AS `user_updated_at`, `user`.`locked_until` AS `user_locked_until`, `user`.`num_failed_attempts` AS `user_num_failed_attempts` FROM `users` `user` WHERE `user`.`email` = ? -- PARAMETERS: ["*********@protonmail.com"]
cache-standalone                     | 1:M 15 Jul 2022 20:18:19.492 * Background saving terminated with success
syncing-server-js-worker-standalone  | {"message":"Alive and kicking!","level":"info"}
auth-worker-standalone               | {"message":"Alive and kicking!","level":"info"}`

**Comment by carlox97** [comment]
> > @moughxyz I'm having the same problem: I self-hosted the docker standalone server and upon the first login after registering I get asked a 2FA code which prevents me from logging in.
> 
> Did you run `./server.sh update` on your server?

> > @moughxyz I'm having the same problem: I self-hosted the docker standalone server and upon the first login after registering I get asked a 2FA code which prevents me from logging in.
> 
> Did you run `./server.sh update` on your server?

Yes, I did. I'm trying to host a local istance (no https) before going public.

```
git clone --single-branch --branch main https://github.com/standardnotes/standalone.git
cd standalone
./server.sh setup

sed -i "s/auth_jwt_secret/$(openssl rand -hex 32)/g" .env
sed -i "s/secret/$(openssl rand -hex 32)/g" docker/auth.env
sed -i "s/legacy_jwt_secret/$(openssl rand -hex 32)/g" docker/auth.env
sed -i "s/secret_key/$(openssl rand -hex 32)/g" docker/auth.env
sed -i "s/server_key/$(openssl rand -hex 32)/g" docker/auth.env

./server.sh update
```

Status looks good, everything running, healtcheck OK.

I tried to register from iOS, Android, Windows using "http://10.0.0.100:3000", I got a generic error at times, upon clicking register. Cannot register with another account because "another registration in progress" (or something in those lines). Upon logging in from mentioned platforms I get prompted the 2FA code, which I don't have, obviously, and at the same time the logs show:

```
auth-standalone                      | query: SELECT `user`.`uuid` AS `user_uuid`, `user`.`version` AS `user_version`, `user`.`email` AS `user_email`, `user`.`pw_nonce` AS `user_pw_nonce`, `user`.`encrypted_server_key` AS `user_encrypted_server_key`, `user`.`server_encryption_version` AS `user_server_encryption_version`, `user`.`kp_created` AS `user_kp_created`, `user`.`kp_origination` AS `user_kp_origination`, `user`.`pw_cost` AS `user_pw_cost`, `user`.`pw_key_size` AS `user_pw_key_size`, `user`.`pw_salt` AS `user_pw_salt`, `user`.`pw_alg` AS `user_pw_alg`, `user`.`pw_func` AS `user_pw_func`, `user`.`encrypted_password` AS `user_encrypted_password`, `user`.`created_at` AS `user_created_at`, `user`.`updated_at` AS `user_updated_at`, `user`.`locked_until` AS `user_locked_until`, `user`.`num_failed_attempts` AS `user_num_failed_attempts` FROM `users` `user` WHERE `user`.`email` = ? -- PARAMETERS: ["************@protonmail.com"]
cache-standalone                     | 1:M 15 Jul 2022 20:18:19.492 * Background saving terminated with success
syncing-server-js-worker-standalone  | {"message":"Alive and kicking!","level":"info"}
auth-worker-standalone               | {"message":"Alive and kicking!","level":"info"}
```

**Comment by pageb018** [comment]
I am still getting the enter 2-fa prompt when trying to create a new account on a brand new install. 

Both backend and app are up to date.

**Comment by carlox97** [comment]
> I am still getting the enter 2-fa prompt when trying to create a new account on a brand new install.
> 
> Both backend and app are up to date.

Try to `./server.sh cleanup` then delete all files and clone from repo again, following the basic setup instructions on their docs and run `./server.sh update`. Then be sure there are no typos in email and pass when you register and try to create a password with a-z A-Z 1-9 and “!#”.
I managed to login without the 2fa error in this way. Still no idea what I was doing wrong.

**Comment by pageb018** [comment]
nvm, I am an idiot. 

I needed to register the account first.
